src/gameState.ts

```diff
@@ -0,0 +1,299 @@
+import {Position} from "./position"
+import {Probability} from "./probability"
+
+export class GameState{
+    public ShipPositions:Array<any>;
+    public MyShots:Array<any>;
+    public OpponentsShots:Array<any>;
+    public remainingShips:number[] = [2,3,3,4,5];  
+    // note that remainingShips always contains the shiplengths in ascending order
+    public getBoard():number[][]{
+        return this.board;
+    }
+    public static letterToNumber:any = {
+        "A":0, 
+        "B":1, 
+        "C":2, 
+        "D":3,
+        "E":4,
+        "F":5,
+        "G":6,
+        "H":7,
+        "I":8,
+        "J":9
+    };
+    public static numberToLetter:any = {
+        0:"A",
+        1:"B",
+        2:"C",
+        3:"D",
+        4:"E",
+        5:"F",
+        6:"G",
+        7:"H",
+        8:"I",
+        9:"J"
+    };
+
+    // 0 means unchecked, 1 means hit,2 means sunk, -1 means missed
+    public board:number[][] = [
+        [0,0,0,0,0,0,0,0,0,0],
+        [0,0,0,0,0,0,0,0,0,0],
+        [0,0,0,0,0,0,0,0,0,0],
+        [0,0,0,0,0,0,0,0,0,0],
+        [0,0,0,0,0,0,0,0,0,0],
+        [0,0,0,0,0,0,0,0,0,0],
+        [0,0,0,0,0,0,0,0,0,0],
+        [0,0,0,0,0,0,0,0,0,0],
+        [0,0,0,0,0,0,0,0,0,0],
+        [0,0,0,0,0,0,0,0,0,0]
+    ];
+
+    // reqBody is the request body, a json object
+    constructor(reqBody:any){
+        // A-J and 1-10
+        this.ShipPositions = (reqBody.ShipPositions)?(reqBody.ShipPositions):[];
+        this.MyShots = (reqBody.MyShots)?(reqBody.MyShots):[];
+        this.OpponentsShots = (reqBody.OpponentsShots)?(reqBody.OpponentsShots):[];
+        if (this.MyShots){
+            for (let i:number = 0; i<this.MyShots.length; i++){
+                let row:number = GameState.letterToNumber[this.MyShots[i].Position.Row];
+                let column:number = this.MyShots[i].Position.Column - 1
+                this.board[row][column] = (this.MyShots[i].WasHit)? 1 : -1;
+            }
+        }
+    }
+
+    public display():void{
+        console.log();
+        console.log(this.board);
+        console.log();
+    }
+
+    public isValidTarget(position:Position):boolean{
+        if (position.row >= 0){
+            if (position.row < 10){
+                if (position.column >= 0){
+                    if (position.column < 10){
+                        return (this.board[position.row][position.column]==0);
+                    }
+                }
+            }
+        }
+        return false;
```

Comment:
Personally, I would structure this as:

```
private isOnBoard(pos: number): boolean {
  let boardMin: number = 0;
  let boardMax: number = 10;
  return pos >= boardMin && pos < boardMax;
}

public isValidTarget(position: Position): boolean {
  if (isOnBoard(position.row) && isOnBoard(position.column) {
    return this.board[position.row][position.column] == 0;
  }
  return false;
}
```

Cuts out a lot of the nasty nesting.